### ISSUE TITLE
Cosmos in Docker + hostname = 💥

### ISSUE THREAD

**Issue by shahkashani** [react-cosmos/react-cosmos#639]
### What's up?

When using Cosmos inside a Docker container, you can't access it via http://localhost:8989, you _have_ to use http://0.0.0.0:8989. 

### Mkay, tell me more...

You can't bind to a hostname in a container and expect to able to access it from the outside. The problem comes from the Express server _always_ being started using a populated `hostname` variable (set to `localhost` by default). If you just make `hostname` undefined by default, it'll run just fine both inside and outside Docker on `localhost`.

You can test this by setting the config `hostname` to `null`, running Cosmos inside and outside a container and making sure http://localhost:8989 works (and note how `hostname: "localhost"` does not work in Docker).

And this would be a totally valid workaround, except you get this in the console:

```
[Cosmos] See you at http://null:8989/
```

Basically, a solution that might is to make hostname undefined by default and changing the console log to something like ```console.log('[Cosmos] See you at http://' + (hostname || 'localhost') + ':' + port + '/');```

**Comment by skidding** [comment]
> You can test this by setting the config hostname to null

How about setting `hostname` to `0.0.0.0`?

**Comment by skidding** [comment]
Yeah, I've been in that situation once and it's ugly.

Until we find a better solution, would you be interested in contributing with a mention about using `0.0.0.0` when running Cosmos inside Docker, in the [Option dump#hostname](https://github.com/react-cosmos/react-cosmos#option-dump) section?

![image](https://user-images.githubusercontent.com/250750/38355025-d2b3c0d8-38e5-11e8-9c80-bcb1d94cc032.png)

It's not much, but I imagine some people will open readme and search for "Docker", and this will at least point them to a working solution. Or encourage them to tell us how to do it better. :)

**Comment by shahkashani** [comment]
Sure, but if you just omit the hostname parameter (or set it to null / undefined) in `app.listen`, it’ll work on localhost both with and without Docker.

Something like `app.listen(port, hostname === ‘localhost’ ? null : hostname)`. Alternatively, which is more work but probably cleaner; set `hostname` to undefined by default in the config and just handle it in the `console.log`, like suggested earlier.

Happy to change the README, though!

**Comment by skidding** [comment]
@shahkashani your solution sounds like a better idea. If you want to make it happen I'm down with it!

So I see two parts to this:
1. Making the default `hostname` undefined (or null, whichever you prefer)
2. As you said, displaying "localhost" when hostname is empty

**Comment by skidding** [comment]
Fixed in [4.5.0](https://github.com/react-cosmos/react-cosmos/releases/tag/v4.5.0). Host is now `null` by default and it will be displayed as a clickable `localhost` link.

**Comment by shahkashani** [comment]
Amazing, thank you so much!
